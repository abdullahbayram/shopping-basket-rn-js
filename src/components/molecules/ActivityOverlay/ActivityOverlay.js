import React from 'react';
import { StyleSheet, View, ActivityIndicator } from 'react-native';
import PropTypes from 'prop-types';

const ActivityOverlay = ({ isVisible, color = 'white', size = 'large' }) => {
  if (!isVisible) return null;

  return (
    <View style={styles.overlay}>
      <ActivityIndicator size={size} color={color} />
    </View>
  );
};

ActivityOverlay.propTypes = {
  isVisible: PropTypes.bool.isRequired,
  color: PropTypes.string,
  size: PropTypes.oneOf(['small', 'large']),
};

export default ActivityOverlay;

const styles = StyleSheet.create({
  overlay: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    backgroundColor: 'rgba(0,0,0,0.5)',
    justifyContent: 'center',
    alignItems: 'center',
    zIndex: 10,
  },
});
