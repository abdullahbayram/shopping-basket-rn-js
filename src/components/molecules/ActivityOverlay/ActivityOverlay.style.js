import { StyleSheet } from 'react-native';
import { sharedColors } from '../../../constants/theme';

export default StyleSheet.create({
  overlay: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    backgroundColor: sharedColors.overlayBackgroundColor,
    justifyContent: 'center',
    alignItems: 'center',
  },
});
