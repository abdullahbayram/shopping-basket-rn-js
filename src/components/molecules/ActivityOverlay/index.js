import React from 'react';
import { View, ActivityIndicator } from 'react-native';
import PropTypes from 'prop-types';
import styles from './ActivityOverlay.style';

const ActivityOverlay = ({ isVisible = false, color = 'white', size = 'large', zIndex = 10 }) => {
  if (!isVisible) return null;

  return (
    <View style={[styles.overlay, { zIndex }]}>
      <ActivityIndicator size={size} color={color} />
    </View>
  );
};

ActivityOverlay.propTypes = {
  isVisible: PropTypes.bool,
  color: PropTypes.string,
  size: PropTypes.oneOf(['small', 'large']),
  zIndex: PropTypes.number,
};

export default ActivityOverlay;
