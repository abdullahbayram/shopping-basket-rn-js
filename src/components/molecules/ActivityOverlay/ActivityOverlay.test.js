import React from 'react';
import { render, screen } from '@testing-library/react-native';
import ActivityOverlay from '.';

describe('ActivityOverlay Component', () => {
  it('renders correctly when isVisible is true', () => {
    const { toJSON } = render(<ActivityOverlay isVisible color="blue" size="small" zIndex={20} />);

    expect(toJSON()).toMatchSnapshot();

    expect(screen.getByTestId('ActivityIndicator')).toBeTruthy();
  });

  it('does not render when isVisible is false', () => {
    const { toJSON } = render(<ActivityOverlay isVisible={false} />);

    expect(toJSON()).toMatchSnapshot();

    expect(screen.queryByTestId('ActivityIndicator')).toBeNull();
  });

  it('applies the correct color and size props to ActivityIndicator', () => {
    render(<ActivityOverlay isVisible color="red" size="large" />);

    const activityIndicator = screen.getByTestId('ActivityIndicator');

    expect(activityIndicator.props.color).toBe('red');
    expect(activityIndicator.props.size).toBe('large');
  });

  it('renders with default props when none are provided', () => {
    render(<ActivityOverlay isVisible />);

    const activityIndicator = screen.getByTestId('ActivityIndicator');

    // Default prop checks
    expect(activityIndicator.props.color).toBe('white');
    expect(activityIndicator.props.size).toBe('large');
  });
});

describe('ActivityOverlay Component - Snapshot Tests', () => {
  it('matches snapshot with default props', () => {
    const { toJSON } = render(<ActivityOverlay isVisible />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('matches snapshot with custom props', () => {
    const { toJSON } = render(<ActivityOverlay isVisible color="green" size="small" zIndex={30} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
