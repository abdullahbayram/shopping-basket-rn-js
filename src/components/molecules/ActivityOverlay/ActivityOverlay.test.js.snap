// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityOverlay Component - Snapshot Tests matches snapshot with custom props 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0,0,0,0.5)",
        "bottom": 0,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
      {
        "zIndex": 30,
      },
    ]
  }
>
  <ActivityIndicator
    color="green"
    size="small"
    testID="ActivityIndicator"
  />
</View>
`;

exports[`ActivityOverlay Component - Snapshot Tests matches snapshot with default props 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0,0,0,0.5)",
        "bottom": 0,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
      {
        "zIndex": 10,
      },
    ]
  }
>
  <ActivityIndicator
    color="white"
    size="large"
    testID="ActivityIndicator"
  />
</View>
`;

exports[`ActivityOverlay Component does not render when isVisible is false 1`] = `null`;

exports[`ActivityOverlay Component renders correctly when isVisible is true 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0,0,0,0.5)",
        "bottom": 0,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
      {
        "zIndex": 20,
      },
    ]
  }
>
  <ActivityIndicator
    color="blue"
    size="small"
    testID="ActivityIndicator"
  />
</View>
`;
